library/src/main/java/org/apache/apex/malhar/lib/wal/FileSystemWAL.java

```diff
@@ -680,9 +681,7 @@ protected void flush() throws IOException
           //until the stream is closed on the local FS, readers don't see any data.
           close();
         } else {
-          Syncable syncableOutputStream = (Syncable)outputStream;
-          syncableOutputStream.hflush();
-          syncableOutputStream.hsync();
+          ((HdfsDataOutputStream)outputStream).hsync(EnumSet.of(SyncFlag.UPDATE_LENGTH));
```

Comment:
The previous code used to work on previous versions of Hadoop, which are still supported with the apex core version that malhar currently depends on. Have you researched if there is a way to fix this without the added dependency? Any pointers on what changed in HDFS to cause this?